### PR TITLE
Upgrade bootstrap to version 4

### PR DESCRIPTION
**Pull request by arthurmcgregor**
#76 improving mobile experience is easier with the newer version of bootstrap.... so I have upgraded most of our pages to use bootstrap 4 👍 

The following pages are still using bootstrap 3:
- setupAudio.pug
- viewAudioRecording.pug
- viewRecording.pug
These are complex pages! Can easily be converted to bootstrap 4 when they have their styling reviewed. To keep them at bootstrap 3, there is an explicit include statement for bootstrap 3 in the head.

All other pages using bootstrap 4 have an explicit include statement for bootstrap 4 in the head, plus a footer at the end of the body which adds the required scripts.

I have not upgraded to the latest version of jQuery (to match the upgraded bootstrap) because it broke Ajax! Not sure if this will affect any bootstrap features or not... seems to be working ok at the moment.

### CODE REVIEW

**Comment by arthurmcgregor** [comment]
Need to restore form validation and error messages to login and register pages

**Comment by mjs** [comment]
@arthurmcgregor I'm happy with the partial approach with converting to Bootstrap v4. Can you please create an issue (or issues) for the pages that still need to be converted so that we're tracking this?

**Comment by mjs** [comment]
@CameronRP can you review this when you have a chance please?

File: views/includes/getQuery.pug [hunk 1]
```diff
@@ -1,62 +1,59 @@
...
-    input#longer-than(type='number')
-    input(type='button', value='Add', onclick="queryUtil.addLongerThan()")
-  .input-group
-    label Shorter than (seconds):
-    input#shorter-than(type='number')
-    input(type='button', value='Add', onclick="queryUtil.addShorterThan()")
-  .input-group
-    label Device ID
-    input#device-id(type='number')
-    input(type='button', value='Add', onClick="queryUtil.addDeviceId()")
```
Comment (CameronRP) [inline views/includes/getQuery.pug]: The new layout for the inputs for "Make new condition" takes up a lot of vertical space on the page. If you could make the label and inputs on the same line this should save a lot of space on the page.

File: views/includes/getQuery.pug [hunk 2]
```diff
@@ -1,62 +1,59 @@
...
-  input(type='text')
-  br
-  label Active query
-  input#active-query(type='text', value='{}')
-  input(type='button', value='Send', onclick="queryUtil.sendQuery()")
-  br
-  label.col-md-2 Limit
-  input#limit.col-md-2(type='number', value=20)
-  label.col-md-2 Offset
-  input#offset.col-md-2(type='number', value=0)
```
Comment (CameronRP) [inline views/includes/getQuery.pug]: If you could make the Limit and Offset inputs a bit wider here so it's easier to see their values.

**Review by CameronRP — CHANGES_REQUESTED** [review]
Looking much nicer. Just these couple of things and also as you said restoring error the messages to login and register pages.

File: views/includes/getQuery.pug (hunk 1 above)
Comment (arthurmcgregor) [reply views/includes/getQuery.pug]: Yup it's on the list of ones to fix better #82. I've reverted it back to the bootstrap 3 version which takes up less vertical space so the page should be usable again. However, I don't want it to hold up the PR so will deal with it later...

File: views/includes/getQuery.pug (hunk 2 above)
Comment (arthurmcgregor) [reply views/includes/getQuery.pug]: Ditto

File: views/includes/getQuery.pug (hunk 1 above)
Comment (arthurmcgregor) [reply views/includes/getQuery.pug]: The idea would be for it to get a similar treatment to the getRecordings.pug page - in fact if the API request functionality is similar, could they be a common component? I'm assuming that was the idea behind a separate getQuery.pug file but seemingly not implemented...

**Review by CameronRP — APPROVED** [review]

**Pull request merged**